src/main/kotlin/com/exactpro/th2/rptdataprovider/server/handler/AbortableRequestHandler.kt

```diff
@@ -36,9 +38,9 @@ internal class AbortableRequestHandler : ChannelInboundHandlerAdapter() {
     }
 
     override fun channelInactive(ctx: ChannelHandlerContext) {
-        val call = ref.getAndSet(null)
-        if (call != null) {
-            call.attributes.getOrNull(ABORT_HANDLER_KEY)?.invoke()
+        ref.getAndSet(null)?.attributes?.getOrNull(ABORT_HANDLER_KEY)?.let {
```

Comment:
note: `let` or `run` should be used when we transform the original object into a new one. If we need to execute additional logic, it is better to use `also`.
However, in this particular case, I think it would be better to have a simple `if` statement

```kotlin
val callback = ref.getAndSet(null)?.attributes?.getOrNull(ABORT_HANDLER_KEY)
if (callback != null) {
    K_LOGGER.debug { "Calling abort callback" }
    callback.invoke()
}
super.channelInactive(ctx)
```